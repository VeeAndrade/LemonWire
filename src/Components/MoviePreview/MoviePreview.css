.movie-poster-image {
  height: 300px;
  width: 200px;
  object-fit: cover;
}

.movie-poster-image:hover {
  cursor: pointer;
  transform: scale(1.06);
  transition: transform .5s;
  box-shadow: 0 0 30px 3px #fbe832;
}

.movie-preview-container {
  margin: 2%;
}

.preview-rating {
  border-radius: 15px;
  font-size: 130%;
  font-weight: bolder;
  height: 20px;
  padding: 12px;
  z-index: 100;
}

.preview-user-rating {
  align-self: flex-end;
}

.rating-lemon {
  position: absolute;
  width: 56px;
  z-index: 90;
}

.user-rating-bundle,
.rating-bundle {
  align-items: center;
  display:flex;
  justify-content: center;
  margin-left: 2%;
}

.user-rating-bundle {
  margin-left: 0%;
  margin-right: 4%;
}

.user-rating-lemon {
  position: absolute;
  width: 56px;
  z-index: 90;
}

.preview-rating-button {
  background-color: #fbe832;
  border: 0;
  border-radius: 3px;
  color: black;
  font-size: 1em;
  font-weight: bold;
  height: 20px !important;
  margin-top: 10px;
  padding: 0;
  position: relative;
  text-decoration: none;
  width: 60px;
  z-index: 100;
}

.preview-rating-button:hover {
  cursor: pointer;
  transform: scale(1.06);
  transition: transform .5s;
  box-shadow: 0 0 10px 1px #fbe832;
}

.ratings {
  align-items: flex-start;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: 0% 5%;
  position: relative;
  top: 15%;
  width: 180px;
}
