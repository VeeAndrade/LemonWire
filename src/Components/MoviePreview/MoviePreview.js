import React from 'react';
import { Link } from 'react-router-dom';
import './MoviePreview.css';
import lemon from '../../Assets/lemon.png';

const MoviePreview = ({ date, movie, saveSelectedMovieToStore  }) => {
  return (
    <article onClick={ () => saveSelectedMovieToStore(movie) } className='movie-preview-container'>
      <p>{date}</p>
      <div className='ratings'>
        <div className='rating-bundle'><p className='preview-rating'>{movie.average_rating}</p>
        <img src={lemon} className='rating-lemon'/></div>
        {movie.user_rating ? <div className='rating-bundle'><p className='preview-user-rating'>{movie.user_rating}</p><img src={lemon} className='rating-lemon'/></div> : <button className='preview-rating-button'>Rate</button>}
      </div>
      <Link to={`/movie_details`}><img src={movie.poster_path} alt='movie poster' className='movie-poster-image'/></Link>
    </article>
  )
}

export default MoviePreview;
