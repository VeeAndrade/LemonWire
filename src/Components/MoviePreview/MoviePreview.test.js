import React from 'react';
import ReactDOM from 'react-dom';
import  MoviePreview from './MoviePreview';
import { shallow } from 'enzyme';
import { Simulate } from 'react-dom/test-utils';

describe('MoviePreview', () => {

  it('should match the snapshot', () => {
    let mockDate = '2020/10/10'
    let mockMovie = {average_rating: 5, poster_path: 'url'};
    const wrapper = shallow(<MoviePreview movie={mockMovie} date={mockDate} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should determine that if the user is logged in, the path is login', () => {
    const mockUserLoggedIn = true;
    let mockDate = '2020/10/10'
    let mockMovie = {average_rating: 5, poster_path: 'url', id: 2};
    const wrapper = shallow(<MoviePreview movie={mockMovie} date={mockDate} userLoggedIn={mockUserLoggedIn}/>);
    const mockDeterminePath = wrapper.instance().determinePath();
    expect(mockDeterminePath).toEqual('/movies/2');
  });

  it('should determine that if the user is logged in, the path is movie details', () => {
    const mockUserLoggedIn = false;
    let mockDate = '2020/10/10'
    let mockMovie = {average_rating: 5, poster_path: 'url', id:2};
    const wrapper = shallow(<MoviePreview movie={mockMovie} date={mockDate} userLoggedIn={mockUserLoggedIn}/>);
    const mockDeterminePath = wrapper.instance().determinePath();
    expect(mockDeterminePath).toEqual('/login');
  });

  it('should fire off saveSelectedMovieToStore with selected movie when rate buttton is clicked', () => {
    let mockSaveSelectedMovieToStore = jest.fn();
    const mockUserLoggedIn = false;
    let mockDate = '2020/10/10'
    let mockMovie = {average_rating: 5, poster_path: 'url', id:2};
    const wrapper = shallow(<MoviePreview movie={mockMovie} saveSelectedMovieToStore={mockSaveSelectedMovieToStore} date={mockDate} userLoggedIn={mockUserLoggedIn}/>);
    wrapper.find('.preview-rating-button').simulate('click')
    expect(wrapper.instance().props.saveSelectedMovieToStore).toHaveBeenCalledWith(mockMovie)
  });
});
