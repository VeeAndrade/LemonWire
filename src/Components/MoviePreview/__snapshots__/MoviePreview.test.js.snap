// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePreview should match the snapshot 1`] = `
<article
  className="movie-preview-container"
  onClick={[Function]}
>
  <p>
    2020/10/10
  </p>
  <div
    className="ratings"
  >
    <div
      className="rating-bundle"
    >
      <p
        className="preview-rating"
      >
        5.0
      </p>
      <img
        className="rating-lemon"
        src="lemon.png"
      />
    </div>
    <Link
      to="/login"
    >
      <button
        className="preview-rating-button"
        onClick={[Function]}
        type="button"
      >
        Rate
      </button>
    </Link>
  </div>
  <Link
    to="/login"
  >
    <img
      alt="movie poster"
      className="movie-poster-image"
      src="url"
    />
  </Link>
</article>
`;
